game/src/js/objects/ui/DOMModal.js

```diff
@@ -56,19 +57,25 @@ export default class DOMModal {
       });
     });
     // Process data into HTML
+    
     if(this.props.data) {
       const objResolve = function(rootObj, string) {
         let value = rootObj;
         let parts = string.split('.');
         while(parts.length != 0) {
           value = value[parts.shift()];
         }
+        
         return value;
       }
+      
       this.modal.querySelectorAll('*[data-value]').forEach(element => {
-        if(element.nodeName == 'INPUT') {
+        if(element.nodeName == 'INPUT' && element.type != 'checkbox') {
           element.value = objResolve(this.props.data, element.getAttribute('data-value'));
         }
+        else if (element.nodeName == 'INPUT' && element.type == 'checkbox') {
+          element.checked = this.props.data.swCheck;
```

Comment:
This should be made generic as the others by taking the value passed in via `data-value` using `objResolve`

`element.checked = objResolve(this.props.data, element.getAttribute('data-value')) == true;`